packages/ui/src/components/va-tabs/VaTabs.vue

```diff
@@ -60,350 +60,401 @@
-          if (this.tabsContentOffset < offsetToSet) {
-            break
-          }
-        }
+    const updateSlider = (tab: TabComponent) => {
+      const tabElement = unref(tab.tabElement)
+      const tabOffsetTop = tabElement?.offsetTop || 0
+      const tabOffsetLeft = tabElement?.offsetLeft || 0
+      const tabClientHeight = tabElement?.clientHeight || 0
+      const tabClientWidth = tabElement?.clientWidth || 0
+
+      if (props.vertical) {
+        const containerClientHeight = container.value?.clientHeight || 0
+        const calculatedSliderOffsetY = containerClientHeight - tabOffsetTop - tabClientHeight
+
+        sliderOffsetY.value = Math.max(calculatedSliderOffsetY, 0)
+        sliderHeight.value = tabClientHeight
+        sliderOffsetX.value = 0
+        sliderWidth.value = 0
+      } else {
+        sliderOffsetX.value = tabOffsetLeft
+        sliderWidth.value = tabClientWidth
+        sliderOffsetY.value = 0
+        sliderHeight.value = 0
       }
     }
 
-    const lastTab = this.context.tabsService?.tabs[this.context.tabsService.tabs.length - 1]
-    const maxOffset = (lastTab?.rightSidePosition || 0) - this.containerRef.clientWidth
+    const includeAnimation = () => {
+      if (!animationIncluded.value) {
+        requestAnimationFrame(() => {
+          animationIncluded.value = true
+        })
+      }
+    }
 
-    offsetToSet = Math.min(maxOffset, offsetToSet)
-    this.tabsContentOffset = Math.max(0, offsetToSet)
+    const redrawTabs = () => {
+      const oldShowPaginationValue = showPagination.value
 
-    this.$emit('click:next')
-  }
+      updatePagination()
 
-  moveToTab (tab: VaTab) {
-    if (this.showPagination && tab.leftSidePosition + this.containerRef.clientWidth <= this.tabsRef.clientWidth) {
-      this.tabsContentOffset = tab.leftSidePosition
-    } else if (this.showPagination && tab.rightSidePosition >= this.containerRef.clientWidth) {
-      this.tabsContentOffset = tab.rightSidePosition - this.containerRef.clientWidth
-    } else {
-      this.tabsContentOffset = 0
+      if (oldShowPaginationValue === showPagination.value) {
+        updateTabsState()
+        includeAnimation()
+      } else {
+        requestAnimationFrame(() => {
+          updateTabsState()
+          includeAnimation()
+        })
+      }
     }
-  }
 
-  updateSlider (tab: VaTab) {
-    if (this.$props.vertical) {
-      const sliderOffsetY = this.containerRef.clientHeight - tab.$el.offsetTop - tab.$el.clientHeight
-      this.sliderOffsetY = Math.max(sliderOffsetY, 0)
-      this.sliderHeight = tab.$el.clientHeight
-      this.sliderOffsetX = 0
-      this.sliderWidth = null
-    } else {
-      this.sliderOffsetX = tab.$el.offsetLeft
-      this.sliderWidth = tab.$el.clientWidth
-      this.sliderOffsetY = 0
-      this.sliderHeight = null
+    const selectTab = (tab: TabComponent) => {
+      if (tab) {
+        valueComputed.value = tab.name || tab.id || tab.id
```

Comment:
Typo

Reply:
Removed typo, reduced nesting. Thank you. :)